src/libraries/Microsoft.Extensions.Hosting.Systemd/tests/UseSystemdTests.cs

```diff
@@ -12,16 +12,28 @@ public class UseSystemdTests
         [Fact]
         public void DefaultsToOffOutsideOfService()
         {
-            var host = new HostBuilder()
+            using IHost host = new HostBuilder()
                 .UseSystemd()
                 .Build();
 
-            using (host)
+            var lifetime = host.Services.GetRequiredService<IHostLifetime>();
+            Assert.IsNotType<SystemdLifetime>(lifetime);
+        }
+
+        [Fact]
+        public void ServiceCollectionExtensionMethodDefaultsToOffOutsideOfService()
+        {
+            var builder = new HostApplicationBuilder(new HostApplicationBuilderSettings
             {
-                var lifetime = host.Services.GetRequiredService<IHostLifetime>();
-                Assert.NotNull(lifetime);
-                Assert.IsNotType<SystemdLifetime>(lifetime);
-            }
+                // Disable defaults that may not be supported on the testing platform like EventLogLoggerProvider.
+                DisableDefaults = true,
+            });
+
+            builder.Services.AddSystemd();
+            using IHost host = builder.Build();
+
+            var lifetime = host.Services.GetRequiredService<IHostLifetime>();
+            Assert.IsNotType<SystemdLifetime>(lifetime);
```

Comment:
What type will it be? If it's actually going to be null, we should validate that instead.

Reply:
It's usually `ConsoleLifetime`. `UseWindowsServiceTests.DefaultsToOffOutsideOfService` does an `Assert.IsType<ConsoleLifetime>(lifetime)` check like you recommend, so I'm not sure why it was different here. Maybe because the `UseSystemdTests` also run on some of the platforms with the `NullLifetime`?

https://github.com/dotnet/runtime/blob/9d0fcb5c2030f2a394eb4d0e675ba0de6cac72d1/src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.netcoreapp.cs#L14-L21

Reply:
That's correct - these tests run on Android, iOS, and WASM. So it it either `ConsoleLifetime` or `NullLifetime`. This test is just making sure it isn't `SystemdLifetime`, since it isn't in a systemd process.

I've added back the `Assert.NotNull`, since as far as I can tell if it is `null`, the test would pass:

https://github.com/xunit/assert.xunit/blob/58a59628853ed7a63895945fdac7984a92da2343/TypeAsserts.cs#L61-L85